### PR TITLE
refactor: improve workflow and images

### PR DESCRIPTION
**Pull request by lukasheinrich**
* adapts permmissions to not require atlas user to run the code
* adapts workflow spec to be less verbose

### CODE REVIEW

File: eventselection/Dockerfile [hunk 1]
```diff
@@ -3,6 +3,8 @@ ADD . /analysis/src
 WORKDIR /analysis/build
 RUN source ~/release_setup.sh &&  \
     sudo chown -R atlas /analysis && \
+    sudo chmod -R 775 /analysis && \
+    sudo chmod -R 755 /home/atlas && \
     cmake ../src && \
     make -j4
 USER root
```
Comment (tiborsimko) [inline eventselection/Dockerfile]: BTW note also one compilation warning when building Docker image:

```
[ 44%] Building CXX object MyEventSelection/CMakeFiles/MyEventSelectionLib.dir/CMakeFiles/MyEventSelectionLibCintDict.cxx.o
/analysis/src/MyEventSelection/Root/MyEventSelectionAlg.cxx: In member function 'virtual EL::StatusCode MyEventSelectionAlg::changeInput(bool)':
/analysis/src/MyEventSelection/Root/MyEventSelectionAlg.cxx:146:57: warning: unused parameter 'firstFile' [-Wunused-parameter]
 EL::StatusCode MyEventSelectionAlg :: changeInput (bool firstFile)
                                                         ^~~~~~~~~
/analysis/src/MyEventSelection/Root/MyEventSelectionAlg.cxx: In member function 'virtual EL::StatusCode MyEventSelectionAlg::execute()':
/analysis/src/MyEventSelection/Root/MyEventSelectionAlg.cxx:208:56: warning: format '%i' expects argument of type 'int', but argument 4 has type 'DataVector<xAOD::Muon_v1>::size_type {aka long unsigned int}' [-Wformat=]
   Info("execute()", "number of muons %i", muons->size());
                                                        ^
[ 55%] Linking CXX shared library ../x86_64-slc6-gcc62-opt/lib/libMyEventSelectionLib.so
Detaching debug info of libMyEventSelectionLib.so into libMyEventSelectionLib.so.dbg
[ 55%] Built target MyEventSelectionLib
```

and also a Python dependency missing:

```
Building wheels for collected packages: hftools
  Running setup.py bdist_wheel for hftools: started
  Running setup.py bdist_wheel for hftools: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/16/1d/30/db7be5c36e96ecae6e5cc930151c24ce77f388ccd378548b48
Successfully built hftools
grin 1.2.1 requires argparse>=1.1, which is not installed.
Installing collected packages: click, brewer2mpl, hftools
Successfully installed brewer2mpl-1.4.1 click-7.0 hftools-0.0.6
```

**Comment by tiborsimko** [comment]
> but then it fails with "Error: Failed to make directory: config":

... but when I keep the workflow definition from `master` branch, then the example works.  So we can commit at least the docker user part and look at simplifying workflow definition later...

**Comment by lukasheinrich** [comment]
@tiborsimko which version of yadage are you using?

**Comment by tiborsimko** [comment]
REANA uses Yadage 0.20.0 and Packtivity 0.14.21, see https://github.com/reanahub/reana-workflow-engine-yadage/blob/master/setup.py#L44-L64

**Pull request merged**